src/components/ContentHighlights/HighlightStepper/SelectContentSelectionCheckbox.jsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { useContextSelector } from 'use-context-selector';
+import PropTypes from 'prop-types';
+import { CheckboxControl } from '@edx/paragon';
+
+import { MAX_CONTENT_ITEMS_PER_HIGHLIGHT_SET } from '../data/constants';
+import { ContentHighlightsContext } from '../ContentHighlightsContext';
+
+const SelectContentSelectionCheckbox = ({ row }) => {
+  const {
+    indeterminate,
+    checked,
+    ...toggleRowSelectedProps
+  } = row.getToggleRowSelectedProps();
```

Comment:
Test coverage will be added in a separate PR here.

Reply:
Leaving open as a reminder to complete